src/features/FreeRooms/FreeRooms.ts

```diff
@@ -0,0 +1,335 @@
+import ROOMS from './Rooms';
+
+async function getRoomSchedHTML(year, semester) {
+    /*
+    Input is an integer representing the year that you'd like to query
+    as well as a string representing the semester for which you'd like
+    to query.
+
+    Possible Semester Values:
+    F - Fall
+    S - Spring
+    A - Summer Session A
+    B - Summer Session B
+    W - Winter Session
+
+    Output is a Promise that, if successful, returns a string of HTML
+    that is received when making a GET request to the
+    Stevens room schedule website.
+    */
+
+    try {
+        const response = await fetch(
+            `https://web.stevens.edu/roomsched?year=${year}&session=${semester}`,
+            {
+                method: 'GET',
+                headers: {
+                    'Content-Type': 'text/html',
+                },
+            }
+        );
+        return await response.text();
+    } catch (e) {
+        console.error(e);
+    }
+}
+
+async function getTablesPerRoom(year, semester) {
+    /*
+    Input is an integer representing the year that you'd like to query
+    as well as a string representing the semester for which you'd like
+    to query.
+
+    Output is an object whose properties are the names of the rooms listed on
+    web.stevens.edu and whose values for those properties are the HTML table elements objects
+    corresponding to those rooms.
+    */
+    let HTMLString;
+    let roomTableDict = {};
+    HTMLString = await getRoomSchedHTML(year, semester);
+    let domparser = new DOMParser();
+    let document = domparser.parseFromString(HTMLString, 'text/html');
+    let roomTables = document.querySelectorAll('table');
+    for (let i = 0; i < ROOMS.length; i++) {
+        //match all of the tables loaded in the HTML to a the name of a room
+        //checking by hand, it seems there are only as many tables as there are rooms
+        //no other tables for decoration or some other data, etc.
+        roomTableDict[ROOMS[i]] = roomTables[i];
+    }
```

Comment:
This construct can usually be replaced with [for..of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) syntax. It's cleaner and usually more intuitive.